client/app/visualizations/chart/plotly/index.js

```diff
@@ -0,0 +1,144 @@
+import { each, debounce, isArray, isObject } from 'underscore';
+
+import Plotly from 'plotly.js/lib/core';
+import bar from 'plotly.js/lib/bar';
+import pie from 'plotly.js/lib/pie';
+import histogram from 'plotly.js/lib/histogram';
+import box from 'plotly.js/lib/box';
+
+import {
+  ColorPalette,
+  prepareData,
+  prepareLayout,
+  calculateMargins,
+  updateDimensions,
+  updateData,
+  normalizeValue,
+} from './utils';
+
+Plotly.register([bar, pie, histogram, box]);
+Plotly.setPlotConfig({
+  modeBarButtonsToRemove: ['sendDataToCloud'],
+});
+
+const PlotlyChart = () => ({
+  restrict: 'E',
+  template: '<div class="plotly-chart-container" resize-event="handleResize()"></div>',
+  scope: {
+    options: '=',
+    series: '=',
+  },
+  link(scope, element) {
+    const plotlyElement = element[0].querySelector('.plotly-chart-container');
+    const plotlyOptions = { showLink: false, displaylogo: false };
+    let layout = {};
+    let data = [];
+
+    const updateChartDimensions = debounce(() => {
+      if (updateDimensions(layout, plotlyElement, calculateMargins(plotlyElement))) {
+        Plotly.relayout(plotlyElement, layout);
+      }
+    }, 100);
```

Comment:
When I pass the optional `immediate` parameter to `debounce` here it stops the flickering (as it doesn't wait the first 100ms). Is there a reason not to do that?

Reply:
Thanks for pointing, I'll focus on this and check it